services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusController.kt

```diff
@@ -395,4 +428,36 @@ class HakemusController(
         logger.warn(ex) { ex.message }
         return HankeError.HAI2006
     }
+
+    @ExceptionHandler(HakemusNotYetInAlluException::class)
+    @ResponseStatus(HttpStatus.CONFLICT)
+    @Hidden
+    fun hakemusNotYetInAlluException(ex: HakemusNotYetInAlluException): HankeError {
+        logger.warn(ex) { ex.message }
+        return HankeError.HAI2013
+    }
+
+    @ExceptionHandler(WrongHakemusTypeException::class)
+    @ResponseStatus(HttpStatus.CONFLICT)
```

Comment:
I wonder whether this error - the given id is for an application that has a type that is not allowed for this operation - is more a 400 error than 409. 

The HTTP 409 Conflict status code indicates a request conflict with the current state of the target resource.

Reply:
I only considered the date as a part of the request, so I considered this to be 409 Conflict. But the id is also a part of the request, so the user has a way to fix their request, which is an indication of 400 Bad Request.

Using 400 now.